jni/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+#
+#   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License").
+#   You may not use this file except in compliance with the License.
+#   A copy of the License is located at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   or in the "license" file accompanying this file. This file is distributed
+#   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+#   express or implied. See the License for the specific language governing
+#   permissions and limitations under the License.
+#
+
+cmake_minimum_required(VERSION 2.8)
+
+project(KNNIndexV1_7_3_6)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+# Target Library to be built
+set(KNN_INDEX KNNIndexV1_7_3_6)
+
+# Check if similarity search exists
+find_path(NMS_REPO_DIR NAMES similarity_search PATHS ${CMAKE_CURRENT_SOURCE_DIR}/external/nmslib REQUIRED)
+
+# If not, pull the updated submodule
+if (NOT EXISTS ${NMS_REPO_DIR})
+    message(STATUS "Could not find nmslib. Pulling updated submodule.")
+    execute_process(COMMAND git submodule update --init -- external/nmslib WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+endif ()
+
+# Add the subdirectory so it is possible to use its targets
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/nmslib/similarity_search)
+
+# Set OS specific variables
+if (${CMAKE_SYSTEM_NAME} STREQUAL Darwin)
+    set(JVM_OS_TYPE darwin)
+    set(LIB_EXT .jnilib)
```

Comment:
I find that .dylib will facilitate debugging jni code. Specifically, one can use "attach to process" in C++ IDE to successfully attach to jps. For .jnilib it will not work for unknown reason. Maybe it is worth building an extra shared library on macOS to facilitate development?

Reply:
Interesting, that would help development. It may be out of the scope of this PR, but is worth investigating. Could you create an issue for it?

Reply:
#87

Reply:
Thanks!